src/scripts/count_refusal.py

```diff
@@ -0,0 +1,414 @@
+        prompt, answer_to_score = create_multiple_choice_prompt_completion(
+            example,
+            cot=False,
+            random_seed=42,
+            permute=False,
+        )
+        score_maps.append(answer_to_score)
+    return prompts, score_maps
+
+def extract_examples(name, datafolder):
+    names = construct_name_to_path_dict(datafolder)
+    if name in names:
+        data_path = names[name]
+        with open(data_path, "r") as file:
+            data = json.load(file)
+
+        return data["examples"]
+
+    else:
+        logger.error(f"Cannot find the name for {name}")
+        return []
+
+def check_llm_fallback(data, datafolder, model):
+    name = data[("name", 0)]
+    _examples = extract_examples(data[("name", 0)], datafolder)
+    if model == "galatica_120b":
+        _mcq_regex_template = MCQ_REGEX_TEMPLATE_2
+        _floatq_regex_template = NUM_REGEX
+    else:
+        _mcq_regex_template = MCQ_REGEX_TEMPLATE_1
+        _floatq_regex_template = FLOATQ_REGEX_TEMPLATE_1 
+
+    completion_text = data["output"]["text"]
+    
+
+    if not np.isnan(data["metrics"]["hamming"]):
+        _metrics = ['multiple_choice_grade']
+        preferred_score = "multiple_choice_grade"
+        if model == "galatica_120b":
+            prompts, score_maps = _prompts_with_choices(_examples)
+        else:
+            prompts, score_maps = _prompts_with_choices_it(_examples)
+    else:
+        _metrics = ['exact_str_match', 'mae', 'mse']
+        preferred_score = "mae"
+
+    llm_extractor = LLMExtractor(
+        model_type="anthropic",
+        seed=42,
+        permute_options=False,
+        temperature=0,
+    )
+    llm_extractor.llm_extraction_count = 0
+    for metric in _metrics:
+        if metric == "multiple_choice_grade":
+            score_map = score_maps[0]
+            prepared_output = prepare_mcq_answer(
+                completion_text,
+                _mcq_regex_template,
+                example=_examples[0],
+                llm_extractor=llm_extractor,
+            )
+        else: 
+            if "target_scores" in _examples[0]:
+                answer = run_regex(
+                    create_multiple_choice_regex(list(score_map.keys())),
+                    prepare_mcq_answer(
+                        completion_text,
+                        _mcq_regex_template,
+                        example=_examples[0],
+                        llm_extractor=llm_extractor,
+                    ),
+                    return_first=False,
+                )
+            else:
+                answer = prepare_general_answer(
+                    completion_text,
+                    pattern=_floatq_regex_template,
+                    llm_extractor=llm_extractor,
+                )
+    
+    return llm_extractor.llm_extraction_count
+
+def count_refusal_error(folder, datafolder, model):
+
+    logger.info(f"Working in model {model}")
+    df = load_all_reports(folder, datafolder)
```

Comment:
is this the slow part? This could then be replaced with `model_scores_dict.pkl`?

Reply:
I do not think so

Reply:
I did not check which part is, but this one is common to other scripts and it is not that slow